code/modules/antagonists/heretic/items/heretic_armor.dm

```diff
@@ -112,17 +111,21 @@
 /obj/item/clothing/suit/hooded/cultrobes/void/proc/hide_item(datum/source, obj/item/item, slot)
 	SIGNAL_HANDLER
 	if(slot & ITEM_SLOT_SUITSTORE)
-		ADD_TRAIT(item, TRAIT_NO_STRIP, REF(src)) // i'd use examine hide but its a flag and yeah
+		item.add_traits(list(TRAIT_NO_STRIP, TRAIT_NO_WORN_ICON), REF(src)) // i'd use examine hide but its a flag and yeah
+		if(isliving(loc))
+			var/mob/living/wearer = loc
+			wearer.update_clothing(slot)
```

Comment:
Adding or removing `TRAIT_NO_WORN_ICON` should handle updating the sprite for you (via a signal_addtrait or something)

Reply:
aight, done in https://github.com/tgstation/tgstation/pull/89343/commits/905d9fade6aa3d5b340efdf8d76d88b0b78f8124